src/components/policy/policy_external/src/policy_manager_impl.cc

```diff
@@ -1871,9 +1903,11 @@ std::string PolicyManagerImpl::ForcePTExchange() {
 
 void policy::PolicyManagerImpl::StopRetrySequence() {
   LOG4CXX_AUTO_TRACE(logger_);
-  if (update_status_manager_.IsUpdateRequired()) {
-    ResetRetrySequence(ResetRetryCountType::kResetWithStatusUpdate);
-  }
+
+  const ResetRetryCountType reset_type =
+      cache_->UpdateRequired() ? ResetRetryCountType::kResetWithStatusUpdate
+                               : ResetRetryCountType::kResetInternally;
+  ResetRetrySequence(reset_type);
```

Comment:
Also irrelevant, what was the purpose of this change?

Reply:
@jacobkeeler unfortunately, it is a part of fix. Without those changes we noticed that the following two scripts are failing:
```
./test_scripts/Policies/App_Permissions/024_ATF_APP_UNAUTHORIZED_If_Failed_Nickname_Validation_After_PTU.lua
./test_scripts/Policies/Validation_of_PolicyTables/245_ATF_preloaded_pt_parameter_trigger_in_local_pt.lua
``
The issue is that in some cases SDL is not sending `UP_TO_DATE` after `UPDATING`.
For example there was an app registered, then PTU in progress (UPDATING) which revokes app permissions, then SDL unregisters this unauthorized app and invokes `StopRetrySequence()` which checks `IsUpdateRequired()` in `UpdateStatusManager` and it returns `true` for UPDATING status, so `ResetRetrySequence` -> `OnResetRetrySequence` will change status to `UPDATE_NEED`, however PTU was finished successfully (`OnValidUpdateReceived()` will be called later).
By that reason, its more reliable to check `update_required` flag in cache manager, which is set right after valid PT update was received. That's why these changes were done in both policy managers.

Please let me know if you need more details about that

Reply:
Hmmm, well is there a reason to call `ResetRetrySequence(ResetRetryCountType::kResetInternally)` if `UpdateRequired` is false?

It seems to me like the change should just be made to the existing `if` statement, rather than adding this new conditional

Reply:
@AKalinich-Luxoft Hmmm, well is there a reason to call `ResetRetrySequence(kResetInternally)` if `UpdateRequired()` is false?

It seems to me like the change should just be made to the existing `if` statement, rather than adding this new conditional:
```cpp
  if (cache_->UpdateRequired()) {
    ResetRetrySequence(ResetRetryCountType::kResetWithStatusUpdate);
  }
```

Reply:
@jacobkeeler hmm, you are right. `kResetInternally` will be called indirectly when PTU is finished with `UP_TO_DATE`. Updated as you suggested in https://github.com/smartdevicelink/sdl_core/pull/3051/commits/f70d036ce4f3e618182ee2e2cf1b0badd38816c8